samza-core/src/main/java/org/apache/samza/execution/JobNodeConfigurationGenerator.java

```diff
@@ -351,10 +351,10 @@ private void configureSerdes(Map<String, String> configs, Map<String, StreamEdge
   private void addSerdes(KV<Serde, Serde> serdes, String streamId, Map<String, Serde> keySerdeMap,
       Map<String, Serde> msgSerdeMap) {
     if (serdes != null) {
-      if (serdes.getKey() != null && !(serdes.getKey() instanceof NoOpSerde)) {
+      if (serdes.getKey() != null) {
         keySerdeMap.put(streamId, serdes.getKey());
       }
-      if (serdes.getValue() != null && !(serdes.getValue() instanceof NoOpSerde)) {
+      if (serdes.getValue() != null) {
```

Comment:
This will impact the maps being returned to the caller which can now have multiple instances of no-op serde even though in theory all of them are logically the same.

While this may not impact functionality, it does create redundant copies of serialized serdes & configurations.

Reply:
The class description suggests we do not want to share references of serdes but even if this creates a NoOp serve copy for each stream, store, table how can it hurt?

Reply:
what class description are you referring to?
Like I pointed out earlier, the harm is creating `copies of serialized serdes & corresponding configurations`. I couldn't find much of historical context here.

@prateekm or @nickpan47  do you guys know why we specifically added this check for `instanceof NoOpSerde` and exclude them?

Reply:
@bharathkk sorry i meant java docs on line 265
```
* We try to preserve the number of Serde instances before and after serialization. However we don't
   * guarantee that references shared between these serdes instances (e.g. an Jackson ObjectMapper shared
   * between two json serdes) are shared after deserialization too.
   *
   * Ideally all the user defined objects in the application should be serialized and de-serialized in one pass
   * from the same output/input stream so that we can maintain reference sharing relationships.
   *
```